### PR TITLE
add pyside 6 support

### PR DESCRIPTION
**Pull request by hannesdelbeke**
removed pyside support, to simplify code

### CODE REVIEW

**Comment by csaez** [comment]
Hi there,

Thanks for the PR!
I see how this is much simpler, but I wonder if this simplification means it cannot run on older versions of the host environment.

When I put this together many years ago, I was using it mostly to reload python tools within a DCC (I used to work on VFX), and it was rare that I could either pull a dependency as PySide into my environment without further side effects (as these environment are tightly controlled) or even be using a recent version of a DCC as pipelines get stuck in time using stable (but old-ish) version of software to de-risk the production.

I would be all in for adding support for pyside6, that would be awesome! but I wouldn't remove support for older host environments for the sake of simplifying this piece of code (which is relatively simple, at least to my eyes).

**Comment by hannesdelbeke** [comment]
i understand that. first i tried to just add pyside 6 to the removed code.
but when testing the old code in maya 2025 i got a strange error.
it was not clear what caused it to me so this was a quicker way to do it. 
and since pyside1 is not used in any dcc anymore these days. it's all pyside 2 or 6, i figured save myself some time.
(Pyside is maya 2016 or older)

**Comment by hannesdelbeke** [comment]
i also just noticed the last few lines were replaced with `show()`, this shouldn't have been committed

**Comment by hannesdelbeke** [comment]
- i re added PySide1 support.
tested in a dcc with PySide6, i don't have access to a PySide1 env.
- restored code at bottom
- upgraded `exec_` to `exec`

File: hard_reload.py [hunk 1]
```diff
@@ -66,8 +60,8 @@ def hard_reload(module_name):
 def show():
     w = HardReload(get_parent())
     w.move_window()
-    w.exec_()
-
+    w.exec()
```
Comment (csaez) [inline hard_reload.py]: question:

Is this compatible with PySide?
If I remember correctly, exec was a reserved keyword on python2 therefore qt developers renamed the method.

Is there any win by renaming? I believe the underscore version has been kept on newer versions for backward compatibility.

File: hard_reload.py [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
-            QtWidgets = _temp.QtWidgets
-        else:
-            QtWidgets = QtGui
+        from PySide2 import QtGui, QtCore, QtWidgets
     except ImportError:
-        pass
-    else:
-        break
+        from PySide import QtGui, QtCore
+        QtWidgets = QtGui
```
Comment (csaez) [inline hard_reload.py:11]: Well done on this simplification, looks much better now!

File: hard_reload.py (hunk 1 above)
Comment (hannesdelbeke) [reply hard_reload.py]: i don't know
exec_ still works but throws a warning in later versions

**Pull request merged**